core/rawdb/database.go

```diff
@@ -148,15 +151,89 @@ func NewDatabase(db ethdb.KeyValueStore) ethdb.Database {
 	}
 }
 
+func ReadOffSetOfCurrentAncientFreezer(db ethdb.KeyValueReader) uint64 {
+	offset, _ := db.Get(offSetOfCurrentAncientFreezer)
+	if offset == nil {
+		return 0
+	}
+	return new(big.Int).SetBytes(offset).Uint64()
+}
+
+func ReadOffSetOfLastAncientFreezer(db ethdb.KeyValueReader) uint64 {
+	offset, _ := db.Get(offSetOfLastAncientFreezer)
+	if offset == nil {
+		return 0
+	}
+	return new(big.Int).SetBytes(offset).Uint64()
+}
+
+func WriteOffSetOfCurrentAncientFreezer(db ethdb.KeyValueWriter, offset uint64) {
+	if err := db.Put(offSetOfCurrentAncientFreezer, new(big.Int).SetUint64(offset).Bytes()); err != nil {
+		log.Crit("Failed to store offSetOfAncientFreezer", "err", err)
+	}
+}
+func WriteOffSetOfLastAncientFreezer(db ethdb.KeyValueWriter, offset uint64) {
+	if err := db.Put(offSetOfLastAncientFreezer, new(big.Int).SetUint64(offset).Bytes()); err != nil {
+		log.Crit("Failed to store offSetOfAncientFreezer", "err", err)
+	}
+}
+
+//NewFreezerDb only create a freezer without statedb.
+func NewFreezerDb(db ethdb.KeyValueStore, frz, namespace string, readonly bool, lastOffSet, newOffSet uint64) (*freezer, error) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	// Create the idle freezer instance, this operation should be atomic to avoid mismatch between offset and acientDB.
+	var frdb *freezer
+	var errors error
+	go func() {
+		frdb, errors = newFreezer(frz, namespace, readonly)
+		if errors != nil {
+			return
+		}
+		frdb.offset = newOffSet
+		offsetBatch := db.NewBatch()
+		WriteOffSetOfCurrentAncientFreezer(offsetBatch, newOffSet)
+		WriteOffSetOfLastAncientFreezer(offsetBatch, lastOffSet)
+		if err := offsetBatch.Write(); err != nil {
+			log.Crit("Failed to write offset into disk", "err", err)
+		}
+		wg.Done()
+	}()
+	if errors != nil {
+		return nil, errors
+	}
+	wg.Wait()
+	return frdb, nil
+}
+
 // NewDatabaseWithFreezer creates a high level database on top of a given key-
 // value data store with a freezer moving immutable chain segments into cold
 // storage.
-func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, readonly bool) (ethdb.Database, error) {
+func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, readonly, disableFreeze bool) (ethdb.Database, error) {
 	// Create the idle freezer instance
 	frdb, err := newFreezer(freezer, namespace, readonly)
 	if err != nil {
 		return nil, err
 	}
+
+	var offset uint64
+	path, name := filepath.Split(freezer)
+	if _, err := os.Stat(filepath.Join(path, "ancient_back")); os.IsNotExist(err) {
```

Comment:
try remove `ancient_back` and `ancient` here.

Reply:
I hardcoded "ancient_back" within the code logic in the same directory tree of old ancient, no need to be specified by user for the new ancientdb directory, so we can assume `ancient_back` as new ancientDB. I removed `ancient` and will use the exact path same to user specified.